File: tp-debug/dummyArg1.f90
    program dummyArg1
!
! number of arguments mismatch program
!
! Rewrite dummyArg1;f90 in Fortran 90 and use module/interface to 
! "prototype" foo.f so that the compiler can detect argument mismatch.
!
      implicit none
      integer i, n
      parameter (n=10)
      real A(n)
      data A/1.,2.,3.,4.,5.,6.,7.,8.,9.,10./

      write(*,*) A

      i = 2
      call foo(A,n)     ! missing third term "i"

    end program dummyArg1

      subroutine foo(A,n,i)
      implicit none
      integer n, i
      real A(n)

      write(*,*)'====> ',i,n,A(i)

    end subroutine foo

File: tp-debug/dummyArg1.c
#include <stdio.h>

/* number of arguments mismatch program. */

/* Modify dummyArg1.c to prototype function foo so that the compiler 
   can detect argument mismatch */

/* void foo(float* A, int* n, int* i); */  /* no prototyping */

#define n 10

int main()
{
  int i;
  float A[n];

  for (i=0; i<n; i++) {
    A[i] = (float) i+1;
  }

  i = 2;
  foo(A, n);   /* missing the third argument */

  return 0;
}


foo.c

void foo(float* A, int n, int i) {
  printf("====> %d, %d, %f\n", i, n, A[i]);
}
